core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1062,8 +1062,8 @@ sealed trait ZIO[-R, +E, +A] extends Serializable with ZIOPlatformSpecific[R, E,
    *   val f3: IO[String, Unit] = f2.resurrectWith(_.getMessage)
    * }}}
    */
-  final def resurrectWith[E1 >: E](f: Throwable => E1): ZIO[R, E1, A] =
-    self.sandbox.mapError(_.failureOrCause.map(c => f(c.squash)).merge)
+  final def resurrectWith[E1 >: E](pf: PartialFunction[Throwable, E1]): ZIO[R, E1, A] =
```

Comment:
I think if we already have an existing combinator that does the same thing we should remove this.

Reply:
Not sure about this, there is `refineWith` and `orDieWith`. The idea was to improve parity/symmetry.

Reply:
In a data type like ZIO that has a lot of structure there are going to be multiple symmetries but I don't think we can have different methods that do the same thing. I think the distinction is that `orDie` / `orDieWith` take all errors and convert them to fiber failures whereas `refine` variants take some errors and convert them to fiber failures. Similarly `resurrect` takes all unchecked errors and converts them to typed errors whereas `unrefineWith` only does that with some errors. So I think that is the appropriate naming.